### DISCUSSION TITLE
RuntimeError: mat1 and mat2 shapes cannot be multiplied (458752x7 and 100352x3)

### DISCUSSION THREAD

**Discussion by YusukeImadegawa** [open-mmlab/mmclassification#1550] [Q&A] [answered]
I changed backborn from VGG18 to ResNet101 cuz VGG is not good perfomance. I got the Error `RuntimeError: mat1 and mat2 shapes cannot be multiplied (458752x7 and 100352x3) ` when I fine-tune with setting the below config.
Input : 224*224 RGB image
Output : 3 chanel
I think # is the cause of error, but i can't set proper one.
# Config
```
_base_ = [
    '../_base_/models/resnet101.py', 
    '../_base_/datasets/imagenet_bs256_davit_224.py', 
    '../_base_/schedules/imagenet_bs1024_adamw_conformer.py', 
    '../_base_/default_runtime.py'
]

model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='ResNet',
        depth=101,
        num_stages=4, #
        in_channels=3, #
        out_indices=(3, ), #
        init_cfg=dict( 
            type='Pretrained',
            checkpoint='https://download.openmmlab.com/mmclassification/v0/resnet/resnet101_8xb32_in1k_20210831-539c63f8.pth',
            prefix='backbone',
        ),
    ),
    neck=None, #
    head=dict(
        type='LinearClsHead',
        num_classes=3,
        in_channels=7*7*2048, #
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
        topk=(1, ),
        cal_acc=True, 
    ))
```
# Structure

+--------------------------+----------------------+-----------+--------------+
| module                   | #parameters or shape | #flops    | #activations |
+--------------------------+----------------------+-----------+--------------+
| model                    | 42.801M              | 7.832G    | 16.232M      |
|  backbone                |  42.5M               |  7.832G   |  16.232M     |
|   backbone.conv1         |   9.408K             |   0.118G  |   0.803M     |
|    backbone.conv1.weight |    (64, 3, 7, 7)     |           |              |
|   backbone.bn1           |   0.128K             |   1.606M  |   0          |
|    backbone.bn1.weight   |    (64,)             |           |              |
|    backbone.bn1.bias     |    (64,)             |           |              |
|   backbone.layer1        |   0.216M             |   0.677G  |   4.415M     |
|    backbone.layer1.0     |    75.008K           |    0.235G |    2.007M    |
|    backbone.layer1.1     |    70.4K             |    0.221G |    1.204M    |
|    backbone.layer1.2     |    70.4K             |    0.221G |    1.204M    |
|   backbone.layer2        |   1.22M              |   1.034G  |   3.111M     |
|    backbone.layer2.0     |    0.379M            |    0.375G |    1.305M    |
|    backbone.layer2.1     |    0.28M             |    0.22G  |    0.602M    |
|    backbone.layer2.2     |    0.28M             |    0.22G  |    0.602M    |
|    backbone.layer2.3     |    0.28M             |    0.22G  |    0.602M    |
|   backbone.layer3        |   26.09M             |   5.191G  |   7.276M     |
|    backbone.layer3.0     |    1.512M            |    0.374G |    0.652M    |
|    backbone.layer3.1     |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.2     |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.3     |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.4     |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.5     |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.6     |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.7     |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.8     |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.9     |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.10    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.11    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.12    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.13    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.14    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.15    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.16    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.17    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.18    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.19    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.20    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.21    |    1.117M            |    0.219G |    0.301M    |
|    backbone.layer3.22    |    1.117M            |    0.219G |    0.301M    |
|   backbone.layer4        |   14.965M            |   0.81G   |   0.627M     |
|    backbone.layer4.0     |    6.04M             |    0.373G |    0.326M    |
|    backbone.layer4.1     |    4.463M            |    0.219G |    0.151M    |
|    backbone.layer4.2     |    4.463M            |    0.219G |    0.151M    |
|  head.fc                 |  0.301M              |           |              |
|   head.fc.weight         |   (3, 100352)        |           |              |
|   head.fc.bias           |   (3,)               |           |              |
+--------------------------+----------------------+-----------+--------------+

==============================
Input shape: (3, 224, 224)
Flops: 7.832G
Params: 42.801M
Activation: 16.232M

**Comment by Ezra-Yu** [comment, marked as answer]
The following is enough fro model setting:
```
_base_ = [
    '../_base_/models/resnet101.py',           # model settings
     ...............
]

# >>>>>>>>>>>>>>> Override model settings here >>>>>>>>>>>>>>>>>>>
model = dict(
    backbone=dict(
        init_cfg=dict( 
            type='Pretrained',
            checkpoint='https://download.openmmlab.com/mmclassification/v0/resnet/resnet101_8xb32_in1k_20210831-539c63f8.pth',
            prefix='backbone',
        ),
    ),
    head=dict(num_classes=3),
)
```
